src/core/SparseMatrix.cpp

```diff
@@ -68,6 +68,20 @@ SparseMatrix::SetZeros(){
     fArmaMat = arma::sp_mat(fNCols, fNRows);
 }
 
+void
+SparseMatrix::SetIdentity(){
+    if(fNRows && fNCols==fNRows )
+        throw DimensionError(Formatter()<<
+                "SparseMatrix:: Can't set identity as matrix is not square. (rows,cols) : ("<<
+                fNRows<<","<<fNCols<<")"  
```

Comment:
I agree, but this condition doesn't catch it. 

To be explicit, if i have `Nrows = 0` `Ncols = 5`:

* `fNRows = false`
* `(fNRows != fNcols) = true` 
* `(fNRows && fNCols != fNRows) =  false` 

So the exception is not thrown even though, as you said, it's not square.